core/divelist.c

```diff
@@ -166,21 +170,39 @@ static int calculate_otu(const struct dive *dive)
 
 	for (i = 1; i < dc->samples; i++) {
 		int t;
-		int po2;
+		int po2i, po2f;
 		struct sample *sample = dc->sample + i;
 		struct sample *psample = sample - 1;
-		t = sample->time.seconds - psample->time.seconds;
+		t = (sample->time.seconds - psample->time.seconds);
 		if (sample->setpoint.mbar) {
-			po2 = sample->setpoint.mbar;
+			po2i = psample->setpoint.mbar;
+			po2f = sample->setpoint.mbar;
 		} else {
 			int o2 = active_o2(dive, dc, psample->time);
-			po2 = lrint(o2 * depth_to_atm(sample->depth.mm, dive));
+			po2i = lrint(o2 * depth_to_atm(psample->depth.mm, dive));	// (initial) po2 at start of segment
+			po2f = lrint(o2 * depth_to_atm(sample->depth.mm, dive));	// (final) po2 at end of segment
+		}
+		if ((po2i > 500) || (po2f > 500)) {			// 1) If PO2 in segment is above 500 mbar
+			if (psample->depth.mm == sample->depth.mm) {	// ... and this is a constant depth segment
+				otu += t * pow(500.0 / ((double)po2i - 500.0), -0.833333333) / 60.0; // otu if time is minutes
```

Comment:
I understand that this is in the logic of the algorithm you looked at... but let's try to understand what "constant depth" actually means when taking into account mm resolution and the precision of depth measurements in the first place.
I think that the condition here is simply random and not all that useful in real life.

Additionally, I'm bothered by the 0.8333333 - that's fake precision. If you want 5/6, then write that, please.

Reply:
will do the 5/6. The real potential problem is a division-by-zero in the equation for ascending/descending dives. I changed the code for this line to "if (po2i == po2f) {" in order to avoid a division by zero. see latest PR version. I understand that acuuracy of depth measurement is not very accurate (my 2 dc's differ by 1.5m in their logging). But for calculation purposes the distinction between a contant-depth segment and an ascending/descending segment is critical.